File: src/talker_zmq.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from std_msgs.msg import String
import zmq

def talker():
    # pub = rospy.Publisher('chatter', String, queue_size=10)
    rospy.init_node('talker_zmq', anonymous=True)
    rate = rospy.Rate(10) # 10hz

    context = zmq.Context()
    socket = context.socket(zmq.PUB)
    socket.bind("tcp://*:6666")

    while not rospy.is_shutdown():
        hello_str = "hello world %s" % rospy.get_time()
        rospy.loginfo(hello_str)
        # pub.publish(hello_str)
        socket.send(hello_str)
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: src/talker_zmq.cpp
/**
 * talker_zmq.cpp
 * Author: Ravi Joshi
 * Date: 2018/01/18
 */

#include <ros/ros.h>

#include <zmq.hpp>

int main(int argc, char** argv) {
  ros::init(argc, argv, "talker_zmq", ros::init_options::AnonymousName);

  ros::Time::init();  // Workaround since we are not using NodeHandle
  ros::Rate loop_rate(10);

  //  Prepare our context and publisher
  zmq::context_t context(1);
  zmq::socket_t publisher(context, ZMQ_PUB);
  publisher.bind("tcp://*:6666");

  while (ros::ok()) {
    std::stringstream msg_str_stream;
    msg_str_stream << "hello_world_" << ros::Time::now();
    std::string msg_str = msg_str_stream.str();

    zmq::message_t message(msg_str.length());
    memcpy(message.data(), msg_str.c_str(), msg_str.length());

    ROS_INFO_STREAM(msg_str);
    publisher.send(message);
    loop_rate.sleep();
  }
  // Clean up your socket and context here
  publisher.close();
  context.close();

  return 0;
}
